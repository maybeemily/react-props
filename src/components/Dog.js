import React from 'react';
import PropTypes from 'prop-types';

function Dog({ name, age, weight, faveFood }) {
  return (
    <ul>
      <li>Name: {name}</li>
      <li>Age: {age}</li>
      <li>Weight: {weight ? weight : 'unknown'}</li>
      <li>Favorite Food: {faveFood ? faveFood : 'unknown'}</li>
    </ul>
  );

}

Dog.propTypes = {
  name: PropTypes.string.isRequired,
  age: PropTypes.number.isRequired,
  weight: PropTypes.string,
  faveFood: PropTypes.string
};

export default Dog;
