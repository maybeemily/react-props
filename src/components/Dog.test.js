import React from 'react';
import { shallow } from 'enzyme';
import Dog from './Dog';

describe('Dog tests', () => {
  it('renders a dog', () => {
    const wrapper = shallow(<Dog name="Saturday" age={16} weight="15 lbs"/>);
    expect(wrapper).toMatchSnapshot();
  });
});
