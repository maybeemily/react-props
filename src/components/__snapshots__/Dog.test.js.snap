// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dog tests renders a dog 1`] = `
<ul>
  <li>
    Name: 
    Saturday
  </li>
  <li>
    Age: 
    16
  </li>
  <li>
    Weight: 
    15 lbs
  </li>
  <li>
    Favorite Food: 
    unknown
  </li>
</ul>
`;
